kotlin/find-route/src/main/java/com/esri/arcgisruntime/sample/findroute/MainActivity.kt

```diff
@@ -0,0 +1,325 @@
+    progressDialog.show()
+
+    val listenableFuture = routeTask.createDefaultParametersAsync()
+    listenableFuture.addDoneListener {
+      try {
+        if (listenableFuture.isDone) {
+          val routeParams = listenableFuture.get()
+          // create stops
+          val stops = arrayListOf(
+            Stop(Point(-117.15083257944445, 32.741123367963446, SpatialReferences.getWgs84())),
+            Stop(Point(-117.15557279683529, 32.703360305883045, SpatialReferences.getWgs84()))
+          )
+          routeParams.apply {
+            setStops(stops)
+            // set return directions as true to return turn-by-turn directions in the route's directionManeuvers
+            isReturnDirections = true
+          }
+
+          // solve the route
+          val result = routeTask.solveRouteAsync(routeParams).get()
+          val route = result.routes[0] as Route
+          // create a simple line symbol for the route
+          val routeSymbol = SimpleLineSymbol(SimpleLineSymbol.Style.SOLID, Color.BLUE, 5f)
+          // create a graphic for the route
+          Graphic(route.routeGeometry, routeSymbol).also {
+            // add the graphic to the map
+            graphicsOverlay.graphics.add(it)
+          }
+
+          if (progressDialog.isShowing) {
+            progressDialog.dismiss()
+          }
+
+          // get the list of direction maneuvers and display it
+          // NOTE: to get turn-by-turn directions the route parameters
+          //  must have the isReturnDirections parameter set to true.
+          val directions = route.directionManeuvers
+          setupBottomSheet(directions)
+
+          // when the route is solved, hide the FAB
+          directionFab.visibility = View.GONE
+        } else {
+          val message = "Error loading route parameters."
+          Log.e(TAG, message)
+          Toast.makeText(this, message, Toast.LENGTH_LONG).show()
+        }
+      } catch (e: Exception) {
+        val message = "Error creating route parameters: ${e.message}"
+        Log.e(TAG, message)
+        Toast.makeText(this, message, Toast.LENGTH_LONG).show()
+      }
+    }
+  }
+
+  /**
+   * Set up the source, destination and route symbols.
+   */
+  private fun setupSymbols() {
+    try {
+      val startDrawable =
+        ContextCompat.getDrawable(this, R.drawable.ic_source) as BitmapDrawable?
+
+      val pinSourceSymbol = PictureMarkerSymbol.createAsync(startDrawable).get()
+      pinSourceSymbol.apply {
+        loadAsync()
+        addDoneLoadingListener {
+          // add a new graphic as start point
+          val sourcePoint = Point(
+            -117.15083257944445,
+            32.741123367963446,
+            SpatialReferences.getWgs84()
+          )
+          // add it to the graphics overlay
+          graphicsOverlay.graphics.add(Graphic(sourcePoint, this))
+        }
+        // make the graphic smaller
+        width = 30f
+        height = 30f
+        offsetY = 20f
+      }
```

Comment:
```suggestion
(ContextCompat.getDrawable(this, R.drawable.ic_source) as? BitmapDrawable)?.let {
        val pinSourceSymbol = PictureMarkerSymbol.createAsync(it).get().apply {
          // make the graphic smaller
          width = 30f
          height = 30f
          offsetY = 20f
        }
        // create a point for the new graphic
        val sourcePoint = Point(
          -117.15083257944445, 32.741123367963446, SpatialReferences.getWgs84()
        )
        // add it to the graphics overlay
        graphicsOverlay.graphics.add(Graphic(sourcePoint, pinSourceSymbol))
      }
```

This is maybe closer to our builder pattern? Happy to hear your thoughts on this!

Reply:
I'm not sure. I feel like because getting the drawable is a fairly complicated line, and this kind of obscures its creation. If we were to use the let{} block, I could perhaps see something like
```
val startDrawable =
        ContextCompat.getDrawable(this, R.drawable.ic_source) as BitmapDrawable?
startDrawable.let{...}
```
But then it's not really doing much, because only the picture marker symbol requires the drawable.

You removed the loadAsync/addDoneLoadingListener inside the picturemarkersymbol.apply block. Are those not necessary because we already called .get()?